core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -19,7 +19,8 @@ package org.apache.spark.util
 
 import java.io.{ByteArrayInputStream, ByteArrayOutputStream}
 
-import scala.collection.mutable.{Map, Set}
+import scala.collection.mutable.{Map, Set, Stack}
+import scala.language.existentials
```

Comment:
Actually it's needed, or you will get compile warning, it's caused by this line: `val (outermostClass, outermostObject) = outerPairs.head`